### ISSUE TITLE
build/contrib/metaphyisicl and build/contrib/timpi are not created oin msys2

### ISSUE THREAD

**Issue by dschwen** [libMesh/libmesh#2447]
The `build/contrib/metaphyisicl` and `build/contrib/timpi` directories should be created and the respective `configure` scripts run. This does not happen on msys2.

This is my current road block after getting PETSc built with all requirements (msmpi, metis, parmetis, hypre).

**Comment by dschwen** [comment]
timpi config uses PETSc's `make getlinklibs` with unfortunately returns a bit of nonsense:

```
$    make PETSC_DIR=/home/daniel/projects/moose/petsc PETSC_ARCH=msys2 getlinklibs
-Wl,-rpath,/home/daniel/projects/moose/petsc/msys2/lib -L/home/daniel/projects/moose/petsc/msys2/lib -Wl,-rpath,/usr/local/petsc/lib -L/usr/local/petsc/lib -Wl,-rpath,/mingw64/lib -L/mingw64/lib -L/home/daniel/projects/moose/petsc/C:msys64mingw64lib -LC:/msys64/mingw64/lib/gcc/x86_64-w64-mingw32/9.2.0 -LC:/msys64/mingw64/lib/gcc -LC:/msys64/mingw64/x86_64-w64-mingw32/lib -LC:/msys64/mingw64/lib -LC:msys64mingw64lib -lpetsc -lHYPRE -lopenblas -lparmetis -lmetis -lmsmpi -lstdc++ -lgfortran -lgcc_s -lquadmath -lm -lgdi32 -luser32 -ladvapi32 -lkernel32 -lquadmath -lstdc++ -lmsmpi -lssp_nonshared -lssp -lmingw32 -lgcc_eh -lmoldname -lmingwex -lmsvcrt -lpthread -ladvapi32 -lshell32 -luser32
```

Note the `-LC:msys64mingw64lib` (slashes missing) and the `-L/home/daniel/projects/moose/petsc/C:msys64mingw64lib` same nonsens bolted onto teh back of a legit path.

**Comment by jwpeterson** [comment]
I'm a little confused, the initial issue comment makes it sound like the build doesn't even get started (can't create directories) but then your first comment makes it seem like timpi config is actually running... Did you already fix the first problem? 

Also I would suspect the issue is something related to git submodules on msys2 since they seem to have that in common.

**Comment by dschwen** [comment]
No, the submodules are in `libmesh/contrib`. The corresponding directories in the build tree are not created (i.e. `libmesh/build/contrib/timpi`).

I have tried manually creating them and running the configure script. My second comment is a bit of an aside. I think in the meantime I have traced the `getlinkpath` issue to https://github.com/msys2/MINGW-packages/issues/6216
